src/dmd/mtype.d

```diff
@@ -5935,6 +5936,7 @@ extern (C++) final class TypeFunction : TypeNext
             }
         }
 
+        auto u = size_t.max; // used for failedIndex if no match
```

Comment:
I would prefer having a `bool failed` and a `size_t failedIndex` and use the `failed` to determine whether or not `failedIndex` holds a valid value.

Reply:
We can just use check the return value against `MATCH.nomatch` to determine if `failedIndex` is valid.